#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 1 0 0 4.71238898038469
  position -0.6899482546172102 2.9382352547256514 -0.9149601630550326
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2.3 2.3
}
Robot {
  translation -0.74 0.03 -0.74
  rotation 0 1 0 1.57
  children [
    DistanceSensor {
      translation 0.065 0.01 0.04
      children [
        Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      lookupTable [
        0 0 0
        2 200 0
      ]
    }
    DistanceSensor {
      translation 0.04 0.01 0.085
      rotation 0 1 0 -1.57
      children [
        Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_fleft"
      lookupTable [
        0 0 0
        2 200 0
      ]
    }
    DistanceSensor {
      translation -0.04 0.01 0.085
      rotation 0 1 0 -1.57
      children [
        Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_fright"
      lookupTable [
        0 0 0
        2 200 0
      ]
    }
    DistanceSensor {
      translation 0.065 0.01 -0.07
      children [
        Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left1"
      lookupTable [
        0 0 0
        2 200 0
      ]
    }
    DistanceSensor {
      translation -0.065 0.01 0.05
      rotation 0 1 0 -3.1415853071795863
      children [
        Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      lookupTable [
        0 0 0
        2 200 0
      ]
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.14 0.01 0.18
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.075 0 -0.03
      }
      device [
        RotationalMotor {
          name "motorL"
        }
      ]
      endPoint Solid {
        translation 0.075 0 -0.03
        rotation 0 0 0.9999999999999999 1.571
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.03
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.075 0 -0.03
      }
      device [
        RotationalMotor {
          name "motorR"
        }
      ]
      endPoint Solid {
        translation -0.075 0 -0.03
        rotation 0 0 0.9999999999999999 1.571
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF casterSolid Solid {
      translation 0 -0.015 0.07
      children [
        DEF caster Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.015
            subdivision 2
          }
        }
      ]
      name "solid(2)"
      boundingObject USE caster
      physics Physics {
      }
    }
  ]
  name "PlexCV"
  boundingObject USE body
  physics Physics {
  }
  controller "plex_maze_controller"
}
Wall {
  translation 0.56 0 0.745
  rotation 0 1 0 1.57
  size 0.37 0.15 0.01
}
Wall {
  translation 0.56 0 0
  rotation 0 1 0 1.57
  name "wall(9)"
  size 0.38 0.15 0.01
}
Wall {
  translation 0.38 0 0.555
  name "wall(1)"
  size 0.37 0.15 0.01
}
Wall {
  translation 0.37 0 -0.555
  name "wall(5)"
  size 0.37 0.15 0.01
}
Wall {
  translation 0.75 0 0.185
  name "wall(7)"
  size 0.37 0.15 0.01
}
Wall {
  translation 0.75 0 -0.185
  name "wall(8)"
  size 0.37 0.15 0.01
}
Wall {
  translation -0.37 0 -0.185
  name "wall(6)"
  size 0.37 0.15 0.01
}
Wall {
  translation -0.185 0 -0.37
  rotation 0 1 0 1.57
  name "wall(10)"
  size 0.38 0.15 0.01
}
Wall {
  translation -0.555 0 0.555
  name "wall(2)"
  size 0.75 0.15 0.01
}
Wall {
  translation -0.185 0 0.935
  name "wall(15)"
  size 1.495 0.15 0.01
}
Wall {
  translation -0.555 0 -0.555
  name "wall(3)"
  size 0.74 0.15 0.01
}
Wall {
  translation -0.93 0 0.185
  rotation 0 1 0 1.57
  name "wall(13)"
  size 1.49 0.15 0.01
}
Wall {
  translation 0 0 -0.935
  name "wall(14)"
  size 1.885 0.15 0.01
}
Wall {
  translation 0.94 0 0
  rotation 0 1 0 1.57
  name "wall(12)"
  size 1.86 0.15 0.01
}
Wall {
  translation -0.185 0 0.185
  name "wall(4)"
  size 0.75 0.15 0.01
}
Wall {
  translation 0.185 0 -0.37
  rotation 0 1 0 1.57
  name "wall(11)"
  size 1.11 0.15 0.01
}
